model/testing/src/icon4py/model/testing/pytest_config.py

```diff
@@ -161,10 +166,23 @@ def pytest_runtest_setup(item):
         item.own_markers,
         grid,
         backend,
-        is_datatest=item.config.getoption("--datatest"),
     )
 
 
+def pytest_collection_modifyitems(config, items):
+    if config.getoption("--datatest-only"):
+        # --datatest-only given in cli: skip all tests that are not marked with @pytest.mark.datatest
+        for item in items:
+            if not [mark.name for mark in item.own_markers if mark.name == "datatest"]:
+                item.add_marker(
```

Comment:
Isn't this the same as `-k 'not datatest'`?

Reply:
hm, you are right.